src/renderer/wg_engine/tvgWgRenderer.cpp

```diff
@@ -221,15 +222,27 @@ bool WgRenderer::postRender()
 }
 
 
-void WgRenderer::dispose(RenderData data)
-{
+void WgRenderer::dispose(RenderData data) {
     auto renderData = (WgRenderDataPaint*)data;
     if (renderData) {
-        if (renderData->type() == Type::Shape)
-            mRenderDataShapePool.free(mContext, (WgRenderDataShape*)renderData);
-        else
-            renderData->release(mContext);
+        ScopedLock lock(mDisposed.key);
+        mDisposed.renderDatas.push(data);
+    }
+}
+
+
+void WgRenderer::clearDisposes()
+{
+    for (uint32_t i = 0; i < mDisposed.renderDatas.count; i++) {
+        auto renderData = (WgRenderDataPaint*)mDisposed.renderDatas[i];
+        if (renderData) {
```

Comment:
Here, the null check is not helpful. Please remove it and keep the code compact and neat.

Reply:
Done